### ISSUE TITLE
After hook is odd to work with

### ISSUE THREAD

**Issue by jeromegn** [cucumber-rs/cucumber#207]
I'm attempting to convert some of our automated testing to use `cucumber-rs`, but I'm finding it hard to tear down my tests.

If the last step fails, the `after` hook doesn't provide a `World` to work with. I need the `World` to get to its state and make decisions on tear down.

My `after` hook:

```rust
#[tokio::main]
async fn main() {
    BalloonWorld::cucumber()
        .max_concurrent_scenarios(1)
        .after(|_, _, _, maybe_world| {
            println!("AFTER! {:?}", maybe_world);
            Box::pin(async move {
                println!("AFTER IN FUT {:?}", maybe_world);
            })
        })
        .run_and_exit("features")
        .await;
}
```

<img width="676" alt="image" src="https://user-images.githubusercontent.com/43325/157325390-8ad0e3ca-667f-4a7c-820f-fba586ecc0a8.png">

What's the recommended way to make this work?

Basically we're creating Fly.io apps and then we need to make sure they are tore down, whatever the outcome of the tests.

**Comment by tyranron** [comment]
@ilslv please, re-check. Once the `World` is initialized, it should be available in Afker Hook.

@jeromegn another option to tear down is to do so in `Drop` for the `World`. We use such a techique a lot in out code bases, and don't use After Hook at all.

**Comment by tyranron** [comment]
@jeromegn released in [0.12.2](https://crates.io/crates/cucumber/0.12.2).
